### DISCUSSION TITLE
LightRAG-Integration

### DISCUSSION THREAD

**Discussion by ByteMeFree** [open-webui/open-webui#6286] [Ideas]
# Feature Request

**Is your feature request related to a problem? Please describe.**
I would like to connect LightRAG (HKUDS/LightRAG) with Open-WebUI. LightRAG is a powerful yet cost-efficient graph database."

**Describe the solution you'd like**
I would like to retrieve context from LightRAG. It is needed to configure some parameters that depend on the query, such as query type (naive, low, high, hybrid, agentic). Offcourse it's alse needed to upload files and update the database.

**Comment by FuncyLec** [comment]
I tried a bit weird way to solve this problem but it works lol. https://github.com/FuncyLec/LightRAG-for-Openwebui

**Reply by ujnxs123** [reply]
How can I set up the LightRAG API server?

**Reply by FuncyLec** [reply]
just as docs said.
https://github.com/HKUDS/LightRAG/blob/main/docs/LightRagAPI.md#api-endpoints

**Reply by bleriot14** [reply]
out of context, are there any better solution than lightrag?

**Reply by FuncyLec** [reply]
> out of context, are there any better solution than lightrag?

@bleriot14 I haven't tried it yet but ，how about https://github.com/HKUDS/MiniRAG

**Reply by bleriot14** [reply]
@FuncyLec  Oh thank you, How could it outperform lightrag. İntresting. Which rag methods did you run before ?

**Reply by ujnxs123** [reply]
> 正如 Docs 所说。 https://github.com/HKUDS/LightRAG/blob/main/docs/LightRagAPI.md#api-endpoints
Thank you for your reply, please tell me how to set the lightrag_server system environment variables, although I started the server but it was run through the * python -m ollama_lightrag_server.py in the api folder, which seems to be different from what is described in the official documentation

**Reply by FuncyLec** [reply]
> > Thank you for your reply, please tell me how to set the lightrag_server system environment variables, although I started the server but it was run through the * python -m ollama_lightrag_server.py in the api folder, which seems to be different from what is described in the official documentation

@ujnxs123 
You can try \examples\lightrag_api_openai_compatible_demo.py , and you can find environment variables in \.env . 
You may need this doc
https://github.com/HKUDS/LightRAG/blob/main/docs/LightRagAPI.md#environment-variables

**Reply by FuncyLec** [reply]
> @FuncyLec Oh thank you, How could it outperform lightrag. İntresting. Which rag methods did you run before ?

Not many, graphrag from microsoft need better PC configuration, not easy for me .
https://microsoft.github.io/graphrag

**Comment by danielaskdd** [comment]
Open WebUI is officially supported now. LightRAG API server can act as an ollama model.  For more information please visit: https://github.com/HKUDS/LightRAG/blob/main/lightrag/api/README.md#about-ollama-api

**Reply by rgaricano** [reply]
https://github.com/HKUDS/LightRAG/blob/main/lightrag/api/README.md#connect-open-webui-to-lightrag

**Reply by GarmischWg** [reply]
this is really nice.

I do look forward to higher level of integration with open webui to enable features like the existing workspace knowledge bank. So that we can use open webui to create seperate graphs and manage documents used to create a graph.

**Reply by andigehle** [reply]
Sadly the link is not working anymore. Is this moved to a more generic doc? Like openai-comp or ollama?

@GarmischWg With the new Event-Functions and own pages, this world be really great to have for xAI / Visuals about the resulting chunks and graphs directly in owui

**Reply by danielaskdd** [reply]
moved to: https://github.com/HKUDS/LightRAG/blob/main/docs/LightRAG-API-Server.md#connect-open-webui-to-lightrag
